pkg/azure/provider/helpers/driver.go

```diff
@@ -239,16 +254,16 @@ func computeDeleteOptionUpdatesForNICsAndDisksIfRequired(resourceGroup string, v
 	return vmUpdateParams
 }
 
-// getNetworkInterfaceReferencesToUpdate checks if there are still NICs which do not have cascade delete set. These are captured and changed
-// NetworkInterfaceReference's are then returned with cascade delete option set.
-func getNetworkInterfaceReferencesToUpdate(networkProfile *armcompute.NetworkProfile) []*armcompute.NetworkInterfaceReference {
+// getNetworkInterfaceReferencesToUpdate checks if there is still the NIC which was created during VM creation with cascade delete not set. It is captured and changed
+// NetworkInterfaceReference is then returned with cascade delete option set.
+func getNetworkInterfaceReferencesToUpdate(networkProfile *armcompute.NetworkProfile, nicToUpdate string) []*armcompute.NetworkInterfaceReference {
 	if networkProfile == nil || utils.IsSliceNilOrEmpty(networkProfile.NetworkInterfaces) {
 		return nil
 	}
 	updatedNicRefs := make([]*armcompute.NetworkInterfaceReference, 0, len(networkProfile.NetworkInterfaces))
 	for _, nicRef := range networkProfile.NetworkInterfaces {
 		updatedNicRef := &armcompute.NetworkInterfaceReference{ID: nicRef.ID}
-		if !isNicCascadeDeleteSet(nicRef) {
+		if *nicRef.ID == nicToUpdate && !isNicCascadeDeleteSet(nicRef) {
```

Comment:
Is this needed? Do we support dynamically attaching NICs to a VM?

Reply:
MCM creates only one nic, but it is possible to attach more using the azure cli.
```
 az vm nic add --help                                                                                                                                                                                                                                    ─╯

Command
    az vm nic add : Add existing NICs to a VM.

Arguments
    --nics              [Required] : Names or IDs of NICs.
    --resource-group -g [Required] : Name of resource group. You can configure the default group
                                     using `az configure --defaults group=<name>`.
    --vm-name           [Required] : The name of the Virtual Machine. You can configure the default
                                     using `az configure --defaults vm=<name>`.
    --primary-nic                  : Name or ID of the primary NIC. If missing, the first NIC in the
                                     list will be the primary.
Example
    Add two NICs to a VM.
        az vm nic add -g MyResourceGroup --vm-name MyVm --nics nic_name1 nic_name2

To search AI knowledge base for examples, use: az find "az vm nic add"

```

Reply:
ok